src/wasm/wrt/ngx_wrt_wasmtime.c

```diff
@@ -624,16 +626,28 @@ ngx_wasmtime_call(ngx_wrt_instance_t *instance, ngx_str_t *func_name,
 
     func = &item.of.func;
 
-    wargs = ngx_pcalloc(instance->pool, sizeof(wasmtime_val_t) * args->size);
-    if (wargs == NULL) {
-        goto done;
+    if (args->size <= 10) {
+        wargs = &swargs[0];
+
+    } else {
+        wargs = ngx_pcalloc(instance->pool,
```

Comment:
It would be great to keep both of these cases covered via the `ngx_wasm_debug` module with a dummy host function interface perhaps.

Reply:
I took a look but I wasn't able to find my way right away... I'm assuming the idea is to add a new `ngx_command_t` from which to call `ngx_wavm_instance_call_func` (or `funcref`?) on a Wasm function with 11 args and 11 rets.

How do I get a `ngx_wavm_instance_t *` from the debug module? And how do I get an `ngx_wavm_func_t *` (or  `ngx_wavm_funcref_t *`?)

I'm assuming I'll need to add a .wat file in the test that declares the 11-arg and 11-ret function; that part I think I can figure out.

Reply:
Both the instance creation and the dummy host function interface will have to be coded into the debug module.

Reply:
I managed to declare the host function interface and write the test case, but I tried to access the `wcf->vm` from `ngx_wasm_core` module to link the module, to avoid having to recreate the whole VM setup, and I think I ended up lost in module initialization ordering shenanigans. I think I might have ended up sinking more time trying to get this test case running than fixing the memory issue. If the lack of coverage is a blocker, I'm inclined to drop the `perf(wasmtime)` commit from this PR to get the `fix(proxy-wasm)` merged.

Reply:
We will do it later then.